com.unity.netcode.gameobjects/Runtime/Core/NetworkManager.cs

```diff
@@ -485,6 +586,399 @@ private void OnValidate()
+                        removeEmptyPrefabs.Add(i);
+                    }
+                }
+
+                // If we have a player prefab, then we need to verify it is in the list of NetworkPrefabOverrideLinks for client side spawning.
+                if (NetworkConfig.PlayerPrefab != null)
+                {
+                    var playerPrefabNetworkObject = NetworkConfig.PlayerPrefab.GetComponent<NetworkObject>();
+                    if (playerPrefabNetworkObject != null)
+                    {
+                        //In the event there is no NetworkPrefab entry (i.e. no override for default player prefab)
+                        if (!NetworkConfig.NetworkPrefabOverrideLinks.ContainsKey(playerPrefabNetworkObject
+                            .GlobalObjectIdHash))
+                        {
+                            //Then add a new entry for the player prefab
+                            var playerNetworkPrefab = new NetworkPrefab();
+                            playerNetworkPrefab.Prefab = NetworkConfig.PlayerPrefab;
+                            NetworkConfig.NetworkPrefabs.Insert(0, playerNetworkPrefab);
+                            NetworkConfig.NetworkPrefabOverrideLinks.Add(playerPrefabNetworkObject.GlobalObjectIdHash,
+                                playerNetworkPrefab);
+                        }
+                    }
+                    else
+                    {
+                        // Provide the name of the prefab with issues so the user can more easily find the prefab and fix it
+                        Debug.LogError($"{nameof(NetworkConfig.PlayerPrefab)} (\"{NetworkConfig.PlayerPrefab.name}\") has no NetworkObject assigned to it!.");
+                    }
+                }
+
+                // Clear out anything that is invalid or not used (for invalid entries we already logged warnings to the user earlier)
+                // Iterate backwards so indices don't shift as we remove
+                for (int i = removeEmptyPrefabs.Count - 1; i >= 0; i--)
+                {
+                    NetworkConfig.NetworkPrefabs.RemoveAt(removeEmptyPrefabs[i]);
+                }
+
+                removeEmptyPrefabs.Clear();
+
+                if (!IsServer)
+                {
+                    State = NetworkManagerState.AwaitingApproval;
+                    if (!NetworkConfig.NetworkTransport.StartClient())
+                    {
+                        Debug.LogError($"Client is shutting down due to network transport start failure of {NetworkConfig.NetworkTransport.GetType().Name}!");
+                        InvokeOnStartupFailedCallback(StartupFailureReason.ConnectFailed);
+                        ShutdownInternal();
+                        return false;
+                    }
+                }
+                else
+                {
+                    if (NetworkConfig.NetworkTransport.StartServer())
```

Comment:
I'm not crazy about having the starting of the transport is embedded in the ResolveNetworkPrefabs function